src/async-wrap.cc

```diff
@@ -407,6 +407,40 @@ static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+class AsyncWrap::DestroyParam {
```

Comment:
I don’t think this needs to be on `AsyncWrap`, it’s kind of it’s own different thing (it can stay in this class though)

Reply:
You mean just dropping up(/down?) into the `node` namespace, but keep it in the same file?

Reply:
Yes, exactly :)